### PR TITLE
fix(stream): arrangement backfill background ddl

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Fixes 2 bugs:
1. Arrangement backfill would not complete previously after recovery as it will be stuck due to incorrect logic.
2. arrangement backfill should not need to commit its internal state table on barrier anymore, after barrier is complete.

Adds background ddl + arrangement backfill as a test step in `main-cron.yml`.

Reduces the logs being uploaded by:
1. Filtering stack traces.
2. Only uploading a zip of the logs.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/executor/backfill/arrangement_backfill.rs [hunk 1]
```diff
@@ -450,54 +451,58 @@ where
                     "backfill_finished_wait_for_barrier"
                 );
                 // If not finished then we need to update state, otherwise no need.
-                if let Message::Barrier(barrier) = &msg
-                    && !is_completely_finished
```
Comment (kwannoel) [inline src/stream/src/executor/backfill/arrangement_backfill.rs:454]: Previously we wait for this flag to pass. However `is_completely_finished` could be true on recovery, if backfill is completed already.

In that case, we should just yield the barrier and stop waiting.

Comment (chenzl25) [reply src/stream/src/executor/backfill/arrangement_backfill.rs:454]: IIUC, after recovery, the backfill state may be inconsistent with the `CreateMaterializedTracker` state, so we need to notify the progress to finish every time after recovery.

File: src/stream/src/executor/backfill/arrangement_backfill.rs [hunk 2]
```diff
@@ -450,54 +451,58 @@ where
...
+                            #[cfg(debug_assertions)]
+                            state_len,
+                            vnodes.iter_vnodes(),
+                        )
+                        .await?;
+                    }
 
                     self.progress
                         .finish(barrier.epoch.curr, total_snapshot_processed_rows);
+                    tracing::trace!(
```
Comment (kwannoel) [inline src/stream/src/executor/backfill/arrangement_backfill.rs:482]: Always notifies.

File: src/stream/src/executor/backfill/arrangement_backfill.rs (hunk 1 above)
Comment (kwannoel) [reply src/stream/src/executor/backfill/arrangement_backfill.rs:454]: Yes, we will always notify see https://github.com/risingwavelabs/risingwave/pull/14563/files#r1464421419. It is outside of the if-else block.

**Review by chenzl25 — APPROVED** [review]

**Pull request merged**